### PR TITLE
Use blazor-lazy-loading

### PR DESCRIPTION
**Pull request by isc30**
Hi there, I managed to use blazor-lazy-loading for module loading :)
The work is based on a feature branch in blazor-lazy-loading cause I'm planning to do some changes based on the work i've done in this PR.

### The purpose of this PR is to demostrate one possible way of using blazor-lazy-loading here

* Substitute custom module loading with blazor-lazy-loading
* New module startup pipeline
* Async download/load of modules (and dependencies)
* Ported Router to LazyRouter
* Changed IoC to DryIoc to register services at runtime

I would make it so there are 2 modulesHost: one for startup (required modules) and another one for lazy modules (optional). Also, considering that this is WebAssembly, you guys need to be more careful with the nuget packages you include. It is obvious that there are too many dependencies on the Theme projects.

Seems like both prerendering and wasm work well, not sure about blazorserver (can anyone check pls?)

I also removed docker cause it didnt work in my PC and pushed by mistake 👯 

Please let me know what you think about this idea :)

Linked issue: isc30/blazor-lazy-loading#41

### CODE REVIEW

File: src/nuget.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <packageSources>
+        <add key="local-repo" value="../../blazor-lazy-loading/output/nuget/" />
```
Comment (isc30) [inline src/nuget.config:4]: this is my local path to the nuget packages 👯

Comment (isc30) [reply src/nuget.config:4]: shouldnt be needed, just delete this line if it gives you errors when building

**Comment by GioviQ** [comment]
Thanks for your contribution. I will learn something new from you.
I would like a true decoupling from every project in the Modules folder.
Moving the module dependencies in new project BlazorBoilerplate.Modules is not the right direction to me.
I am finding a way to implement autodiscovery of module dll in Modules folder.

**Comment by isc30** [comment]
This PR is **not** using the manifest system that blazor-lazy-loading provides. You can use the manifest system to discover modules and their content without loading the DLL.
There is no way to automatically discover static dll files in wasm directly. You will need to create a controller in the server that informs the client about this (similar as _lazy.json) but during runtime.

This can be easily implemented by using a custom IManifestLocator, do you want to work more in this direction?

**Comment by GioviQ** [comment]
First of all I have to check how it works https://github.com/oqtane/oqtane.framework

**Comment by isc30** [comment]
It works downloading a json from the server

https://github.com/oqtane/oqtane.framework/blob/master/Oqtane.Client/Services/ModuleService.cs#L24

The same I already implemented, but with a controller and custom path
instead of a static _lazy.json

The path for these 2 can be easily changed in IModuleLocator, and
IAssemblyLocator

>

**Comment by GioviQ** [comment]
Some modules can be installed from https://www.nuget.org/packages?q=oqtane
This is the more interesting part.

**Comment by isc30** [comment]
You can also have NuGet modules with the approach I described. It's the same having a ProjectReference or a PackageReference.
I don't think I follow your exact needs here.

**Comment by GioviQ** [comment]
Oqtane has UI to browse nuget.org, download and install packages runtime. There is not a static reference on the packages you can find on https://www.nuget.org/packages?q=oqtane
Maybe I don't understand your approch, but Oqtane seems different.

**Comment by enkodellc** [comment]
@isc30 this looks very good. I am going to let @GioviQ work with you on to flush out this functionality.

**Comment by isc30** [comment]
Is [this](https://github.com/dotnet/aspnetcore/issues/5465#issuecomment-628436194) what you want to build?
Do you want me to update the PR to hit AspNetCore endpoints so you can build anything you want in the backend?

**Comment by GioviQ** [comment]
At the moment I am working on other aspects of the project.
Thanks

**Comment by isc30** [comment]
What if you tell me if the approach I linked is what you want?
Since it's almost finished, I don't see the value of rejecting the PR. I can change that and get it merged.

**Comment by GioviQ** [comment]
As I said in an old comment, I prefer the implemetation of Oqtane modules. No lazy loading at all.
Modules and themes installation from a gallery.

**Comment by GioviQ** [comment]
@isc30 Sorry. I changed my mind about it. It's my fault. 😃

**Comment by isc30** [comment]
Oqtane modules use lazy loading to download and load the DLLs into memory (not as part of the boot). This can be easily achieved by using a custom AssemblyLocator.

**Pull request closed without merging**